src/yki/boundary/cas.clj

```diff
@@ -30,14 +34,40 @@
                                      :headers      {"Caller-Id" caller-id}})))
 
 (defn- json-request [^String method url data]
-  (let [json-str        (json/write-str data)
+  (let [body            (when data (json/write-str data))
         request-builder (RequestBuilder. method)]
     (doto request-builder
       (.addHeader "Content-Type" "application/json")
-      (.setBody json-str)
+      (.setBody body)
       (.setUrl url))
     (.build request-builder)))
 
+(defn- clear-ticket-stores! [^CasClient cas-client]
+  (let [cls                   (.getClass cas-client)
+        session-fetcher-field (.getDeclaredField cls "casSessionFetcher")
+        _                     (.setAccessible session-fetcher-field true)
+        session-fetcher       (.get session-fetcher-field cas-client)]
+    (.clearTgtStore ^CasSessionFetcher session-fetcher)
+    (.clearSessionStore ^CasSessionFetcher session-fetcher)))
+
+(defn- refresh-tickets-response? [url-helper {:keys [headers status]}]
+  (and (= 302 status)
```

Comment:
and -> if ?

Reply:
Sit pitäisi olla joku `false`-branchikin.

Reply:
Tää oli ainakin itselle vähän hankalalukuinen ja tajusin vasta nyt että tuo `when-let + str/starts-with?` oli `and:n toinen parametri`. Katsoin alkuun että tää toteutus oli tyyppiä `(and (= 302 status)` ja jos true, niin palauta when-let:n paluuarvo.

Käviskö about tällainen ratkaisu tilalle:
```
(defn- refresh-tickets-response? [url-helper {:keys [headers status]}]
  (let [is-redirect       (= 302 status)
        matching-location (when-let [location (.get ^HttpHeaders headers "Location")]
                            (str/starts-with? location (url-helper :cas.login.root)))]
    (and is-redirect matching-location)))
```

Reply:
> Katsoin alkuun että tää toteutus oli tyyppiä (and (= 302 status) ja jos true, niin palauta when-let:n paluuarvo.

Ei tää alkuperäinen intuitio tästä mun mielestä väärä oo:
```
(defmacro and
  "Evaluates exprs one at a time, from left to right. If a form
  returns logical false (nil or false), and returns that value and
  doesn't evaluate any of the other expressions, otherwise it returns
  the value of the last expr. (and) returns true."
...)
```

Nythän ehkä isoimpana erona tuo ehdottamasi muutos siis evaluoi tuon `(when-let [location ...] ...)` lausekkeen turhaan silloin jos kyseessä ei ole redirect.

Reply:
Joo, toki kun `matching-location` ei ole "lazy" muuttuja, mutta sellaisia ei taida saada clojuressa tehtyäkään. Mutta voi kai tää jäädä kuten tää toistaiseksi onkin, itselle oli silti alkuun hankalalukuinen vähemmällä clojure-osaamisella.